perftestnotebook.py

```diff
@@ -144,66 +158,92 @@ def process(self):
 
         self.fmt_data = fmt_data
 
+        """
+        Write formatted data output to filepath
+        """
+        print(json.dumps(self.fmt_data, indent=4, sort_keys=True))
+
+        prefix = "output" if "prefix" not in self.config else self.config["prefix"]
+        output_data_filepath = "%s_fmt_data.json" % prefix
+
+        if "output" in self.config:
+            output_data_filepath = self.config["output"]
+
+        print("Writing results to %s" % output_data_filepath)
+
+        with open(output_data_filepath, "w") as f:
+            json.dump(self.fmt_data, f, indent=4, sort_keys=True)
+
+        """
+        Analyze data
+        """
         if "analysis" in self.config:
             # Analyze the data
-            all_results = {}
-            self.analyzer.data = fmt_data
-            for func in self.config["analysis"]:
-                all_results[func] = getattr(self.analyzer, func)()
-            return all_results
+            # TODO
+            # something like: 
+            # for analysis_method in analysis:
+            #   analyzer.process(analysis_method,self.fmt_data)
+            pass
 
-        return self.fmt_data
+        """
+        Post to iodide server
+        """
+        if args.no_iodide:
+            pass
+        else:
+            self.post_to_iodide(output_data_filepath)
 
+        return self.fmt_data
 
-def main():
-    args = parse_args()
+    def post_to_iodide(self, output_data_filepath):
+        # Upload template file to Iodide
+        template = "testing/template/template.txt"
```

Comment:
As I mentioned above, we shouldn't be displaying the template.txt file. We need to build a template.txt file and display that one.

I imagine that this would have two stages to start. It would start with inserting some preamble, like a generic title of the report as well as the fetch for the data, and the next stage would add the code to run in the notebook.